### PR TITLE
Add style for smallcaps.

### PR DESCRIPTION
**Pull request by Zimmi48**
This is a fix for the rendering in HTML of the smallcaps in the manual. Unfortunately, we cannot merge it as is because it makes the inconsistent use of `Coq` vs `|Coq|` visible, including in the introduction.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
With #12172 merged can this proceed?

**Comment by Zimmi48** [comment]
Note that I fixed most uses of `Coq` or `OCaml` but not those of `Gallina` or `Ltac`.

File: doc/sphinx/changes.rst [hunk 1]
```diff
@@ -4134,13 +4134,13 @@ General infrastructure
...
   particular, the default is now that warnings are printed by coqc.
-- In asynchronous mode, Coq is now capable of recovering from errors and
+- In asynchronous mode, |Coq| is now capable of recovering from errors and
   continue processing the document.
 
 Tools
 
 - coqc accepts a -o option to specify the output file name
-- coqtop accepts --print-version to print Coq and OCaml versions in
+- coqtop accepts --print-version to print |Coq| and OCaml versions in
```
Comment (jfehrle) [inline doc/sphinx/changes.rst:4143]:
```suggestion
- coqtop accepts --print-version to print |Coq| and |OCaml| versions in
```

File: doc/sphinx/changes.rst [hunk 2]
```diff
@@ -5701,31 +5701,31 @@ Extraction
...
+|CoqIDE|
 
-- Coqide now runs coqtop as separated process, making it more robust:
+- |CoqIDE| now runs coqtop as separated process, making it more robust:
   coqtop subprocess can be interrupted, or even killed and relaunched
   (cf button "Restart Coq", ex-"Go to Start"). For allowing such
   interrupts, the Windows version of coqide now requires Windows >= XP
   SP1.
-- The communication between CoqIDE and Coqtop is now done via a dialect
+- The communication between |CoqIDE| and Coqtop is now done via a dialect
```
Comment (jfehrle) [inline doc/sphinx/changes.rst:5711]:
```suggestion
- The communication between |CoqIDE| and coqtop is now done via a dialect
```

File: doc/sphinx/changes.rst [hunk 3]
```diff
@@ -6823,7 +6823,7 @@ Extraction
   not happen anymore.
 
 - The command Extract Inductive has now a syntax for infix notations. This
-  allows in particular to map Coq lists and pairs onto Caml ones:
+  allows in particular to map |Coq| lists and pairs onto Caml ones:
```
Comment (jfehrle) [inline doc/sphinx/changes.rst:6826]:
```suggestion
  allows in particular to map |Coq| lists and pairs onto |OCaml| ones:
```

right?

File: doc/sphinx/history.rst [hunk 4]
```diff
@@ -1171,7 +1171,7 @@ Incompatibilities
 
 - New naming strategy for NewInduction/NewDestruct may affect 7.1 compatibility
 - Extra parentheses may exceptionally be needed in tactic definitions.
-- Coq extensions written in Ocaml need to be updated (see dev/changements.txt
+- |Coq| extensions written in Ocaml need to be updated (see dev/changements.txt
```
Comment (jfehrle) [inline doc/sphinx/history.rst:1174]:
```suggestion
- |Coq| extensions written in |OCaml| need to be updated (see dev/changements.txt
```

File: doc/sphinx/language/core/primitive.rst [hunk 5]
```diff
@@ -45,13 +45,13 @@ applications of these primitive operations.
 
 The extraction of these primitives can be customized similarly to the extraction
 of regular axioms (see :ref:`extraction`). Nonetheless, the :g:`ExtrOCamlInt63`
-module can be used when extracting to OCaml: it maps the Coq primitives to types
-and functions of a :g:`Uint63` module. Said OCaml module is not produced by
+module can be used when extracting to |OCaml|: it maps the |Coq| primitives to types
+and functions of a :g:`Uint63` module. Said |OCaml| module is not produced by
```
Comment (jfehrle) [inline doc/sphinx/language/core/primitive.rst:49]:
```suggestion
and functions of a :g:`Uint63` module. That |OCaml| module is not produced by
```

"Said" is mostly used for legal documents, overly formal for most other texts.

File: doc/sphinx/language/extensions/match.rst [hunk 6]
```diff
@@ -639,7 +639,7 @@ Dependent pattern matching
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The examples given so far do not need an explicit elimination
-predicate because all the |rhs| have the same type and Coq
+predicate because all the |rhs| have the same type and |Coq|
```
Comment (jfehrle) [inline doc/sphinx/language/extensions/match.rst:642]: Is the `|rhs|` correct?

File: doc/sphinx/practical-tools/coqide.rst [hunk 7]
```diff
@@ -153,7 +153,7 @@ as standard |GtkSourceView| styles are available. Other styles can be
 added e.g. in ``$HOME/.local/share/gtksourceview-3.0/styles/`` (see
 the general documentation about |GtkSourceView| for the various
 possibilities). Note that the style of the rest of graphical part of
-Coqide is not under the control of |GtkSourceView| but of GTK+ and
+|CoqIDE| is not under the control of |GtkSourceView| but of GTK+ and
```
Comment (jfehrle) [inline doc/sphinx/practical-tools/coqide.rst:156]: `|GtkSourceView|` is correct?

File: doc/sphinx/proof-engine/ltac2.rst [hunk 8]
```diff
@@ -4,7 +4,7 @@ Ltac2
 =====
 
 The Ltac tactic language is probably one of the ingredients of the success of
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/ltac2.rst:6]:
```suggestion
The |Ltac| tactic language is probably one of the ingredients of the success of
```

File: doc/sphinx/proof-engine/ltac2.rst [hunk 9]
```diff
@@ -4,7 +4,7 @@ Ltac2
...
-Coq, yet it is at the same time its Achilles' heel. Indeed, Ltac:
+|Coq|, yet it is at the same time its Achilles' heel. Indeed, Ltac:
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/ltac2.rst:7]:
```suggestion
|Coq|, yet it is at the same time its Achilles' heel. Indeed, |Ltac|:
```

File: doc/sphinx/proof-engine/ltac2.rst [hunk 10]
```diff
@@ -538,7 +538,7 @@ is a proper one or referring to something in the Ltac context.
 
 Likewise, in Ltac1, constr parsing is implicit, so that ``foo 0`` is
 not ``foo`` applied to the Ltac integer expression ``0`` (Ltac does have a
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/ltac2.rst:540]:
```suggestion
not ``foo`` applied to the Ltac integer expression ``0`` (|Ltac| does have a
```

**Review by jfehrle — APPROVED** [review]
Looks good.  A few details to address.  And get the refman to build.

**Comment by jfehrle** [comment]
One other thought: how about having coqbot merge check whether everything in the build succeeded?  I have to remind myself to check that before merging.  coqbot could require "ignore failures" to ignore failing jobs.

File: doc/sphinx/language/extensions/match.rst (hunk 6 above)
Comment (Zimmi48) [reply doc/sphinx/language/extensions/match.rst:642]: It's a short-hand defined in this particular file:

```
.. |rhs| replace:: right hand sides
```

File: doc/sphinx/practical-tools/coqide.rst (hunk 7 above)
Comment (Zimmi48) [reply doc/sphinx/practical-tools/coqide.rst:156]: It's defined in this particular file as:

```
.. |GtkSourceView| replace:: :smallcaps:`GtkSourceView`
```

**Comment by Zimmi48** [comment]
> One other thought: how about having coqbot merge check whether everything in the build succeeded? I have to remind myself to check that before merging. coqbot could require "ignore failures" to ignore failing jobs.

That's possible of course. I didn't do that so far because of how often there are some spurious failures.

File: doc/sphinx/language/extensions/match.rst (hunk 6 above)
Comment (jfehrle) [reply doc/sphinx/language/extensions/match.rst:642]: Hardy seems worth defining the symbol.  It's only used in 2 places in a single paragraph and it requires mental translation when you're looking at the `rst` file.  And you have to remember that it's "right hand side**s**", not "right hand side".  Just a mild rant, don't change it unless you want to.

**Comment by Zimmi48** [comment]
The build is passing now and your suggestions have been applied @jfehrle.

**Comment by jfehrle** [comment]
Is there any significance to the MacOS failure?

@coqbot: merge now

**Comment by Zimmi48** [comment]
>Is there any significance to the MacOS failure?

This is a spurious failure that has happened frequently for a few weeks. Cf. https://github.com/Homebrew/homebrew-core/issues/61772 and https://coq.zulipchat.com/#narrow/stream/237656-Coq-devs.20.26.20plugin.20devs/topic/macOS.20CI

**Comment by Zimmi48** [comment]
Bummers! We should have checked the rendering. It turns out that `|Coq|` inside a `** **` context or a link is not substituted.
Furthermore, the PDF output does not show anything in small caps!

I'm now wondering why we cared about this smallcaps style in the first place. It doesn't even look good IMO.

Anyway, I'm not backporting this PR. I'm wondering if we shouldn't simply revert it and remove the definitions of `|Coq|`, `|OCaml|`, and everything else that is defined as smallcaps.

cc @cpitclaudel

**Comment by Zimmi48** [comment]
FTR, here is the current master refman where we can see the issue of stuff not being substituted: https://coq.github.io/doc/master/refman/

And here is the PDF artifact of this PR where we can see that there is no style for smallcaps: https://coq.gitlab.io/-/coq/-/jobs/800869705/artifacts/_install_ci/share/doc/coq/sphinx/latex/CoqRefMan.pdf

**Comment by cpitclaudel** [comment]
> I'm now wondering why we cared about this smallcaps style in the first place. It doesn't even look good IMO.

I'm with you on this one. (And on cursing against reStructuredText's handling of nested markup)

**Comment by jfehrle** [comment]
Looks like the style was present in the 8.7 PDF, but not in 8.9+.  8.8 wasn't downloadable so I didn't check.

image

IMHO smallcaps looks better with serif fonts.  But we're using sans-serif in the HTML version:

image

**Pull request merged**